linux/devdoc/threadpool_linux_requirements.md

```diff
@@ -237,7 +237,7 @@ MOCKABLE_FUNCTION(, int, threadpool_schedule_work, THANDLE(THREADPOOL), threadpo
 
 **SRS_THREADPOOL_LINUX_07_048: [** If reallocating the task array fails, `threadpool_schedule_work` shall fail and return a non-zero value. **]**
 
-**SRS_THREADPOOL_LINUX_07_049: [** `threadpool_schedule_work` shall copy the work function and work function context into insert position in the task array and return zero on success. **]**
+**SRS_THREADPOOL_LINUX_07_049: [** `threadpool_schedule_work` shall initialize `pending_work_item_count_ptr` with `NULL` then copy the work function and work function context into insert position in the task array and return zero on success. **]**
```

Comment:
>and return zero on success [](http://example.com/codeflow?start=214&length=26)

This part is not needed since you have essentially the same spec on line 248

Reply:
I have seen this all over the place. Looks like line 240 wants to specify a jump to line 248 or some kind of a branch around 240.

Reply:
I agree with Jelani. Putting a "return zero on success" clause in this requirement makes it sound like we are return right away without touching the subsequent specs. Seems like we go on to release a lock so that is probably not what we want here.

Reply:
I would say that the "and assign `0` to the return variable to indicate success" clause is not really required. There is no a good way you can write a test for this, and we don't really do this in other places.